Fix build on non-PowerPC non-Intel systems (e.g. Apple Silicon). Fixes:

pc_resource.c:1542:21: error: use of undeclared identifier 'PDF_PLATFORM'
--- libs/pdcore/pc_config.h.orig	2012-06-06 06:58:58.000000000 -0500
+++ libs/pdcore/pc_config.h	2021-01-12 12:59:53.000000000 -0600
@@ -142,8 +142,7 @@
 
 /* try to identify the Mac OS X command line compiler */
 
-#if defined(__APPLE__) && (defined(__ppc__) || \
-    defined(__i386__) || defined(__ppc64__) || defined(__x86_64__))
+#if defined(__APPLE__) && defined(__MACH__)
 
 #define MACOSX
 
